kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -572,14 +598,7 @@ public void run() {
                             }
                         }
 
-                        publishInternal(message);
-
-                        // TODO: add a 'message throttle' configuration parameter to
-                        // slow down publish rate?
-
-                        // Notify the listeners
-                        DataServiceImpl.this.dataServiceListeners.onMessagePublished(message.getId(),
-                                message.getTopic());
+                        tryPublishInternal(message);
```

Comment:
Because if there aren't tokens in the bucket, the Data service can't publish the message. So, it tries to publish something...